File: 0541-reverse-string-ii/0541-reverse-string-ii.cpp
class Solution {
public:
    string sub(string s, int start, int end, bool flip = false) {
        string res = "";

        for(int j = start; j < end; j++) {
            if(j < s.size()) res = flip ? s[j] + res : res + s[j];
        }

        return res;
    }

    string reverseStr(string s, int k) {
        string res = "";

        for(int i = 0; i < s.size(); i+=2*k) { 
            res += sub(s, i, i+k, true);
            res += sub(s, i+k, i+2*k);
        }

        return res; 
    }
};

File: 0541-reverse-string-ii/0541-reverse-string-ii.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var reverseStr = function(s, k) {
    let res = "";
    
    for(let i = 0; i < s.length; i+=2*k) { 
        let str = s.substring(i, i+k);
        
        let l = str.length-1;
        for(let j = 0; j < str.length; j++) {
            res += str[l--];
        }
        res += s.substring(i+k, i+2*k);
    }

    return res;
};
